scripts/create-release-archive.js

```diff
@@ -0,0 +1,193 @@
+  )
+
+  console.log(workdir)
+
+  copyReleaseFiles(repoDir, workdir, { prefix: name, ref: ref })
+
+  // Make the changes we want to make
+  // Currently just removing dev stuff from package.json
+  console.log('Updating package.json')
+  updatePackageJson(path.join(workdir, name, 'package.json'), cleanPackageJson)
+
+  // Create the release archive in the project root (or destDir)
+  zipReleaseFiles({
+    cwd: workdir, file: path.join(argv.dest || repoDir, releaseArchive), prefix: name
+  })
+
+  console.log(`Saved release archive to ${argv.dest ? argv.dest : ''}${releaseArchive}`)
+
+  // Clean up
+  fs.rmdirSync(workdir, { recursive: true })
+}
+
+function updatePackageJson (file, updater) {
+  let pkg
+  pkg = JSON.parse(fs.readFileSync(file, { encoding: 'utf8' }))
+  pkg = updater(pkg)
+  fs.writeFileSync(file, JSON.stringify(pkg, null, 2) + os.EOL, { encoding: 'utf8' })
+  // update package-lock.json to match
+  child_process.execSync('npm install', { cwd: path.dirname(file), encoding: 'utf8', stdio: 'inherit' })
+}
+
+function cleanPackageJson (pkg) {
+  // remove dev dependencies
+  delete pkg.devDependencies
+
+  // remove config for dev dependencies
+  delete pkg.jest
+  delete pkg.standard
+
+  // remove dev scripts
+  pkg.scripts = Object.fromEntries(
+    Object.entries(pkg.scripts)
+      .filter(([name]) => packageJsonScriptsInclude.includes(name))
+  )
+
+  return pkg
+}
+
+function getReleaseVersion (ref) {
+  if (!ref) {
+    const packageVersion = require('../package').version
+    if (isNewVersion(packageVersion)) {
+      return packageVersion
+    }
+  }
+
+  ref = ref || 'HEAD'
+
+  const versionString = child_process.execSync(`git describe --tags ${ref}`, { encoding: 'utf8' }).trim()
+  const version = versionString.slice(1) // drop the initial 'v'
+
+  return version
+}
+
+function isNewVersion (version) {
+  return !!child_process.spawnSync(
+    'git', ['rev-parse', `v${version}`]
+  ).status
+}
+
+function copyReleaseFiles (src, dest, { prefix, ref }) {
+  // We are currently using the export-ignore directives in .gitattributes to
+  // decide which files to include in the release archive, so the easiest way
+  // to copy all the release files is `git archive`
+  child_process.execSync(
+    `git archive --format=tar --prefix="${prefix}/" ${ref} | tar -C ${dest} -xf -`,
+    { cwd: src }
+  )
+}
+
+function zipReleaseFiles ({ cwd, file, prefix }) {
+  zipCreate(
+    {
+      cwd: cwd,
+      file: file,
+      exclude: path.join(prefix, 'node_modules', '*')
+    },
+    prefix
+  )
+}
+
+function zipCreate ({ cwd, file, exclude }, files) {
+  files = Array.isArray(files) ? files : [files]
+
+  const zipArgs = ['--exclude', exclude, '-r', file, ...files]
+  const ret = child_process.spawnSync(
+    'zip', zipArgs,
+    { cwd: cwd, encoding: 'utf8', stdio: 'inherit' }
+  )
+  if (ret.status !== 0) {
+    throw `zip \\\n\t${zipArgs.join(' \\\n\t')}\n:Failed with status ${ret.status}` // eslint-disable-line
```

Comment:
Small things:

1. Should we use `throw new Error()` instead of throwing a string?
2. Can we do `['zip'].concat(zipArgs).join('\\\n\t')` to avoid the repetition of the new line/tab string?

Reply:
> Should we use throw new Error() instead of throwing a string?

Throwing an Error results in a less helpful traceback for some reason :(

> Can we do ['zip'].concat(zipArgs).join('\\\n\t') to avoid the repetition of the new line/tab string?

What about `['zip', ...zipArgs].join('\\\n\t')`?